src/pipecat/services/resembleai/tts.py

```diff
@@ -0,0 +1,286 @@
+        # 3 model, and it's worth it for the better audio quality.
+        # We also don't want to automatically push LLM response text frames,
+        # because the context aggregators will add them to the LLM context even
+        # We can use those to generate text frames ourselves aligned with the
+        # playout timing of the audio!
+        super().__init__(
+            aggregate_sentences=True,
+            push_text_frames=False,
+            pause_frame_processing=True,
+            sample_rate=sample_rate,
+            text_aggregator=text_aggregator or SkipTagsAggregator([("<spell>", "</spell>")]),
+            **kwargs,
+        )
+
+        self._api_key = api_key
+        self._url = url
+        self._voice_uuid = voice_uuid
+        self._output_format = output_format
+        self._precision = precision
+        self._binary_response = binary_response
+        self._no_audio_header = no_audio_header
+        
+        self._websocket = None
+        self._receive_task = None
+        self._request_id = 0
+        self._current_audio_buffer = bytearray()
+        self._is_receiving_audio = False
+
+    def can_generate_metrics(self) -> bool:
+        return True
+
+    def _build_msg(self, text: str = "", request_id: Optional[int] = None) -> str:
+        if request_id is None:
+            request_id = self._request_id
+            self._request_id += 1
+            
+        msg = {
+            "voice_uuid": self._voice_uuid,
+            "data": text,
+            "binary_response": self._binary_response,
+            "request_id": request_id,
+            "output_format": self._output_format,
+            "sample_rate": self.sample_rate,
+            "precision": self._precision,
+            "no_audio_header": self._no_audio_header,
+        }
+        return json.dumps(msg)
+
+    async def start(self, frame: StartFrame):
+        await super().start(frame)
+        await self._connect()
+
+    async def stop(self, frame: EndFrame):
+        await super().stop(frame)
+        await self._disconnect()
+
+    async def cancel(self, frame: CancelFrame):
+        await super().cancel(frame)
+        await self._disconnect()
+
+    async def _connect(self):
+        await self._connect_websocket()
+
+        if self._websocket and not self._receive_task:
+            self._receive_task = self.create_task(self._receive_task_handler(self._report_error))
+
+    async def _disconnect(self):
+        if self._receive_task:
+            await self.cancel_task(self._receive_task)
+            self._receive_task = None
+
+        await self._disconnect_websocket()
+
+    async def _connect_websocket(self):
+        try:
+            if self._websocket and self._websocket.open:
+                return
+            logger.debug("Connecting to Resemble")
+
+            self._websocket = await websockets.connect(self._url, extra_headers={"Authorization": f"Bearer {self._api_key}"})
```

Comment:
Connect using the websocket asyncio client:

```suggestion
            self._websocket = await websocket_connect(self._url, extra_headers={"Authorization": f"Bearer {self._api_key}"})
```